atomic_reactor/inner.py

```diff
@@ -330,6 +331,7 @@ def base_image_inspect(self):
         return self._base_image_inspect
 
     def throw_canceled_build_exception(self, *args, **kwargs):
+        self.build_canceled = True
```

Comment:
Let's add a test to make sure this behavior doesn't accidentally break in the future.